external/netutils/netlib_netmon.c

```diff
@@ -0,0 +1,188 @@
+			if (sock->remote_port == 0) {					\
+				printf("%-8s", "*");						\
+			} else {										\
+				printf("%-8d", sock->remote_port);			\
+			}												\
+		} else {											\
+			printf("IPv4\t");								\
+			printf("%-15s:", inet_ntoa(sock->local_ip));	\
+			if (sock->local_port == 0) {					\
+				printf("%-8s", "*");						\
+			} else {										\
+				printf("%-8d", sock->local_port);			\
+			}												\
+			printf("%-15s:", inet_ntoa(sock->remote_ip));	\
+			if (sock->remote_port == 0) {					\
+				printf("%-8s", "*");						\
+			} else {										\
+				printf("%-8d", sock->remote_port);			\
+			}												\
+		} 													\
+	} while (0)
+#else
+#define PRINT_IPV(sock)									\
+	do {												\
+		printf("IPv4\t");								\
+		printf("%-15s:", inet_ntoa(sock->local_ip));	\
+		if (sock->local_port == 0) {					\
+			printf("%-8s", "*");						\
+		} else {										\
+			printf("%-8d", sock->local_port);			\
+		}												\
+		printf("%-15s:", inet_ntoa(sock->remote_ip));	\
+		if (sock->remote_port == 0) {					\
+			printf("%-8s", "*");						\
+		} else {										\
+			printf("%-8d", sock->remote_port);			\
+		}												\
+	} while (0)
+#endif
+
+
+/**
+ * Print a externally used socket information.
+ */
+static void print_socket(sq_queue_t *q_sock)
+{
+	int i = 1;
+	struct netmon_sock *sock_info = (struct netmon_sock *) sq_remfirst(q_sock);
+	printf("\nSockfd\tProto\tType\tLocal\t\t\tRemote\t\t\tstate\tPID\n");
+	while (sock_info) {
+		printf("%-8d",i);
+		/* Protocol */
+		if (sock_info->type & NETMON_TCP) {
+			printf("TCP\t");
+			PRINT_IPV(sock_info);
+		} else if (sock_info->type & NETMON_UDP) {
+			printf("UDP\t");
+			PRINT_IPV(sock_info);
+		} else {
+			printf("RAW\t");
+			PRINT_IPV(sock_info);
+		}
+
+		switch (sock_info->state) {
+			case NETMON_WRITE:
+				printf("WRITE\t");
+				break;
+			case NETMON_LISTEN:
+				printf("LISTEN\t");
+				break;
+			case NETMON_CONNECT:
+				printf("CONNECT\t");
+				break;
+			case NETMON_CLOSE:
+				printf("CLOSE\t");
+				break;
+			default:
+				printf("NONE\t");
+				break;
+		}
+		printf("%d:%s\n", sock_info->pid, sock_info->pid_name);
+		i++;
+		free(sock_info);
+		sock_info = (struct netmon_sock *) sq_remfirst(q_sock);
+	}
+
+	return ;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netlib_netmon_sock
+ *
+ * Description:
+ *   Get the socket info
+
+ * Parameters:
+ *   command   Type of information to get
+ *
+ * Return:
+ *   0 on success; -1 on failure
+ *
+ ****************************************************************************/
+
+int netlib_netmon_sock(void)
+{
+	int ret = ERROR;
+	sq_queue_t q_sock;
+	sq_init(&q_sock);
+	/* Get sockets */
+
+	int sockfd = socket(PF_INETX, NETLIB_SOCK_IOCTL, 0);
```

Comment:
This socket is not related to any transport sockets(IPv4, IPv6) so I think PF_INETX is not proper.

Reply:
That is a quite true. Modification completed.